### ISSUE TITLE
Applicability tests don't properly guard against execution

### ISSUE THREAD

**Issue by shanman190** [openrewrite/rewrite#2746]
Given the following:
```
type: specs.openrewrite.org/v1beta/recipe
name: com.example.rewrite.gradle.UpdateGradleWrapper
displayName: Update the Gradle version, conditionally
description: Update the Gradle version, conditionally.
applicability:
  singleSource:
    - org.openrewrite.gradle.search.FindGradleWrapper:
        version: (,6)
        distribution: bin
recipeList:
  - org.openrewrite.gradle.UpdateGradleWrapper:
      version: 6.9.3
      distribution: bin
```

When given a repository with a Gradle wrapper less than 6.0, I would expect this recipe to execute. However, it also improperly executes on repositories that have a Gradle wrapper >= 6.0 as well.

**Comment by sambsnyd** [comment]
I went and found other repositories this reproduces one. Here is one such example, chosen because it is a very small repository:
https://github.com/crotwell/version-class

Reproducing the issue locally using this exact yaml and rewrite-gradle-plugin 5.34.0, I notice this in the console output:
```
> Task :rewriteDryRun
Validating active recipes
Recipe validation error in distributionType: must be a valid distribution type
Recipe validation error in distributionType: must be a valid distribution type
Recipe validation error in distributionType: must be a valid distribution type
Recipe validation errors detected as part of one or more activeRecipe(s). Execution will continue regardless.
```

Now the yaml you have provided looks completely valid, the indentation is correct and "bin" is a perfectly valid value for the distribution field on that recipe. So maybe there's a problem in YamlResource loader or Environment where it pertains to constructing recipes to be applicability tests

**Comment by shanman190** [comment]
I don't disagree that the validation error is something to be fixed as well, but the absence of the distribution type still results in the same behavior including the error messaging stating that distribution type must be valid.

```
type: specs.openrewrite.org/v1beta/recipe
name: com.shelter.rewrite.gradle.UpdateGradleWrapper
displayName: Update to the current recommended Gradle version
description: Update to the current recommended Gradle version.
applicability:
  singleSource:
    - org.openrewrite.gradle.search.FindGradleWrapper:
        version: (,6)
recipeList:
  - org.openrewrite.gradle.UpdateGradleWrapper:
      version: 6.9.3
      distribution: bin
  - com.shelter.rewrite.gradle.MigrateGradleEnterprise
```

```
> Task :rewriteDryRun
Validating active recipes
Recipe validation error in distributionType: must be a valid distribution type
Recipe validation error in distributionType: must be a valid distribution type
Recipe validation error in distributionType: must be a valid distribution type
Recipe validation errors detected as part of one or more activeRecipe(s). Execution will continue regardless.
```

**Comment by kunli2** [comment]
Fixes #2676

**Comment by kunli2** [comment]
This issue should have been fixed by PR #2754.
